### ISSUE TITLE
Initialize moving averages by the first applied value rather than 0

### ISSUE THREAD

**Issue by sdemyanov** [tensorflow/tensorflow#2825]
Currently hidden moving averages are initialized by 0. If the decay is large, it takes a lot of time for the moving average variable to forget about initial zero value. It would be much more reasonable to initialize by the first obtained value of the observed variable, and then start to apply decay. In this case the initial initialization is not required anymore, so the variable can be removed from the "assert_variables_initialized" list

**Comment by cssndrx** [comment]
I can't reproduce what you're seeing. Could you add a small code snippet to repro? 

```
tf.reset_default_graph()
sess = tf.InteractiveSession()

x = tf.Variable(3.0)
ema = tf.train.ExponentialMovingAverage(0.9)
maintain_averages_op = ema.apply([x])

sess.run(tf.initialize_all_variables())
sess.run(x.assign(x+1.0))
sess.run(maintain_averages_op)
print 'x', sess.run(x)
print 'avg', sess.run(ema.average(x))

```

This returns:
```
x 4.0
avg 3.1
```

**Comment by golmschenk** [comment]
@sdemyanov is probably using a Tensor in the moving average rather than a Variable, since the documentation for `tf.train.ExponentialMovingAverage.apply` specifies that "For Tensor objects, the shadow variables are initialized to 0."

**Comment by tarikarici** [comment]
Yes, I have the same problem with @sdemyanov. Doing batch normalization using tf.nn.moments and initializing to zero as a big problem in my case.

**Comment by aselle** [comment]
Automatically closing due to lack of recent activity. Please reopen when further information becomes available.

**Comment by CosmosShadow** [comment]
the same question.

**Comment by aselle** [comment]
@tarikarici, @sdemyanov, @CosmosShadow, did you try @golmschenk's suggestion? Could one of you provide a small reproducing snippet if you want us to look at it further.

**Comment by prb12** [comment]
Closing due to lack of activity.

**Comment by wielandbrendel** [comment]
I am running into the same problem. Here is a minimum example:

```
tf.reset_default_graph()
sess = tf.InteractiveSession()

x = tf.identity(tf.Variable(3.0))
ema = tf.train.ExponentialMovingAverage(0.9)
maintain_averages_op = ema.apply([x])

sess.run(tf.initialize_all_variables())
sess.run(maintain_averages_op)
print('x', sess.run(x))
print('avg', sess.run(ema.average(x)))
```
This returns
```
x 3.0
avg 0.3
```

**Comment by sdemyanov** [comment]
Guys, the issue was resolved by zero_debias=True parameter. Just use that one.

**Comment by wielandbrendel** [comment]
@sdemyanov Thanks! I completely misunderstood this parameter.

**Comment by ozabluda** [comment]
`zero_debias=True` mitigates this issue after a couple of hundred iterations, but has almost no effect after few iterations.
